File: 0066-plus-one/solution.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry = 0;
        for (int i = digits.size()-1; i >= 0; i--) {
            int sum = digits[i] + carry + ((i == digits.size()-1) ? 1 : 0);
            if (sum >= 10)
                carry = sum / 10;
            else 
                carry = 0;
            digits[i] = sum % 10;
        }

        if (carry) digits.insert(digits.begin() + 0, carry);
        return digits;
    }
};

File: 0066-plus-one/solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        for i in reversed(range(len(digits))):
            s = digits[i] + carry
            digits[i] = s % 10
            carry = s // 10
        if carry:
            digits = [1] + digits
        return digits

